Aligned README file after separating extensions (#30)
openSUSE/suse-vscode-doc: README.md

https://github.com/openSUSE/suse-vscode-doc -- feel free to add your suggestions
via GitHub pull requests or issues.

## Extensions

Many tasks can be automatized by using *extensions*. They are available from
VSCode marketplaces:
* for VSCode: https://marketplace.visualstudio.com/VSCode
* for VSCodium: https://open-vsx.org/

The documentation team delivers the following extension for VSCode:
* DocBook Snippets
* SUSE Vale rules

> TIP: you can install them from VSCode by pressing CTRL+SHIFT+X and typing the

Reviewer: Is it only for VSCode? Or does it also works on VSCodium? If both works, maybe use a neutral word:

```suggestion
TIP: you can install them from the editor by pressing CTRL+SHIFT+X and typing the
```

### Support for RELAX NG schema

> To let VSCode be aware of GeekoDoc and DocBook assembly schema, you have two

Reviewer: Shouldn't it be plural?

```suggestion
To let VSCode be aware of GeekoDoc and DocBook assembly schemas, you have two
```

### Resolve references inside the current file

> To autocomplete XML IDs from the current file in `<xref linkend=""/>`

Reviewer: ```suggestion
To autocomplete DocBook XML IDs from the current file when using `<xref linkend=""/>`
```

> references, add the following to `settings.json`:

Reviewer: ```suggestion
references, add the following lines to `settings.json`:
```

To ignore specific strings such as XML markup from being spell-checked, insert
matching regular expression to `settings.json`:

    ```json
    "cSpell.ignoreRegExpList": [
        "/<(\n|.)*?>/g",
        "/<\/.*?>/g",
        "/&.*;/g",
        "/<(screen|command|package|option|filename|systemitem).*?>(\n|.)*?</(\\1)>/g",
    ],
    ```

### Marking as incorrect

You can insert a list of phrases that are always considered incorrectly spelled.
Add the list to  `settings.json`:

    ```json
    "cSpell.flagWords": [
        "filesystem"
    ],
    ```

## DocBook snippets

Snippets are templates of complete DocBook structures. You can insert them into
a document by typing defined prefixes. This is an example of a `<para>` snippet:

    ```json
    "Paragraph": {
        "scope": "xml",
        "prefix": "pa",
        "body": [
            "<para>",
            "\t${1:insert text}",
            "</para>"
        ]
    },
    ```

> You can obtain all the snippets collection by installing the *DocBook Snippets*
> extension. Alternatively, clone the

Reviewer: Just an idea:

```suggestion
Obtain all snippets by installing the *DocBook Snippets* extension.
Alternatively, clone the
```

https://github.com/openSUSE/vscode-docbook-snippets repository and make a
symbolic link in your `~/.config/VSCodium/User/snippets` directory, for example:

    ```bash

>     ln -s /home/USER_NAME/devel/vscode-extensions/vscode-docbook-snippets/snippets/snippets.code-snippets docbook.code-snippets

Reviewer: You can shorten it a bit:

```suggestion
    ln -s ~/devel/vscode-extensions/vscode-docbook-snippets/snippets/snippets.code-snippets docbook.code-snippets
```

The following task example formats the open XML file with the `daps-xmlformat`
command:

    ```json
    "label": "Daps: XML format",
    "type": "shell",
    "command": "daps-xmlformat",
    "args": [
        "-i",
        "${file}"
    ],
    "group": "build",
    "presentation": {
        "echo": true,
        "reveal": "silent",
        "panel": "shared",
    },
    ```

To assign a shortcut to this particular task, create a `keybindings.json` file
in your user configuration directory add the following content:

    ```json
    [
        {
            "key": "ctrl+shift+i",
            "command": "workbench.action.tasks.runTask",
            "args": "Daps: XML format",
            "when": "editorLangId == xml"
        },
    ]
    ```

TIP: because the XML extension already defines `ctrl+shift+i`, disable XML
formatting in `tasks.json`:

    ```json
    "xml.format.enabled": false,
    ```

See https://github.com/openSUSE/suse-vscode-doc for my collection of daps tasks.

## Style checker

Vale style checker can apply rules from SUSE styleguide and highlight errors and
warning in the text after savin the file with CTRL+S. To make it work, you need

> to install the Vale binary and either install the *SUSE Vale rules* extension
> into VSCode, or configure Vale manually as described in

Reviewer: I would split the sentences as it's a bit confusing what comes first and what's optional.

```suggestion
to install the Vale binary first. Then either install the *SUSE Vale rules* extension
into VSCode or configure Vale manually as described in
```
